src/components/Calendar/index.js

```diff
@@ -12,7 +12,81 @@ export default function Calendar({
   capture,
   isAutosized
 }) {
-  const [{ pinnedCrns }] = useContext(TermContext);
+  const [{ pinnedCrns, oscar }] = useContext(TermContext);
+
+  const dayMap = DAYS.reduce((acc, day) => {
+    acc[day] = {};
+    return acc;
+  }, {});
+
+  const crns = [...new Set([...pinnedCrns, ...(overlayCrns || [])])];
+  const meetingLen = (m) => m.period.end - m.period.start;
+  crns.sort(
+    (a, b) =>
+      Math.max(
+        ...oscar
+          .findSection(a)
+          .meetings.filter((m) => m.period)
+          .map(meetingLen)
+      ) -
+      Math.max(
+        ...oscar
+          .findSection(b)
+          .meetings.filter((m) => m.period)
+          .map(meetingLen)
+      )
+  );
+
+  crns.forEach((crn) => {
```

Comment:
Can you add some comments explaining the overview of the algorithm used here? It seems to be both iterative and recursive and use a visited set, but the exact mechanism is unclear.